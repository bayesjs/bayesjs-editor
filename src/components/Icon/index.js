import React from 'react';
import PropTypes from 'prop-types';
import fontAwesome from 'font-awesome/css/font-awesome.css';
import classNames from 'classnames';
import { concat } from 'ramda';

const Icon = ({ name }) =>
  <i className={classNames(fontAwesome.fa, fontAwesome[concat('fa-', name)])} />;

Icon.propTypes = {
  name: PropTypes.string.isRequired,
};

export default Icon;
