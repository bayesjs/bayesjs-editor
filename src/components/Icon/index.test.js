import React from 'react';
import { render } from '@testing-library/react';
import Icon from './index';

const renderComponent = (props) => {
  const { container } = render(
    <Icon {...props} />,
  );

  return container.firstChild;
};

describe('Icon Component', () => {
  const defaultProps = {
    name: 'search',
  };
  let component;

  beforeEach(() => {
    component = renderComponent(defaultProps);
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
