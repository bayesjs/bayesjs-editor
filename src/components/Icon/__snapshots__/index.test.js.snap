// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component matches snapshot 1`] = `
<i
  class="fa fa-search"
/>
`;
